File: src/_03_BehavioralPattern/_03_05_Mediator/java/after/FrontDesk.java
package _03_BehavioralPattern._03_05_Mediator.java.after;

public class FrontDesk {

  private CleaningService cleaningService = new CleaningService(this);

  private Restaurant restaurant = new Restaurant(this);

  public String getRoomNumberFor(Integer guestId) {
    return "방번호 123";
  }

  public void getTowers(Guest guest, int numberOfTowers) {
    this.cleaningService.getTowers(guest.getId(), numberOfTowers);
  }

  public void dinner(Guest guest, String menu) {
    this.restaurant.dinner(guest.getId(), menu);
  }
  
}

File: src/_03_BehavioralPattern/_03_05_Mediator/FrontDesk.js
import CleaningService from "./CleaningService.js";
import Restaurant from "./Restaurant.js";

/**
 * @typedef { import("./Guest").default } Guest
 */

export default class FrontDesk {
  /** @type { CleaningService } */
  _cleaningService;

  /** @type { Restaurant } */
  _restaurant;
  
  constructor() {
    this._cleaningService = new CleaningService(this);
    this._restaurant = new Restaurant(this);
  }

  /** @returns { string } */
  getRoomNumber(guestId) {
    return `🐫🐫🐫 - ${guestId}`;
  }
  
  /**
   * @param { Guest } guest 
   * @param { number } numberOfTowers 
   */
  getTowers(guest, numberOfTowers) {
    this._cleaningService.getTowers(guest.getId(), numberOfTowers);
  }

  /**
   * 
   * @param { Guest } guest 
   * @param { string } menu 
   */
  dinner(guest, menu) {
    this._restaurant.dinner(guest.getId(), menu);
  }
}